src/sst/core/ssthandler.h

```diff
@@ -0,0 +1,168 @@
+// Copyright 2009-2021 NTESS. Under the terms
+// of Contract DE-NA0003525 with NTESS, the U.S.
+// Government retains certain rights in this software.
+//
+// Copyright (c) 2009-2021, NTESS
+// All rights reserved.
+//
+// This file is part of the SST software package. For license
+// information, see the LICENSE file in the top level directory of the
+// distribution.
+
+#ifndef SST_CORE_SSTHANDLER_H
+#define SST_CORE_SSTHANDLER_H
+
+namespace SST {
+
+// This file contains base classes for use as various handlers (object
+// encapsulating callback functions) in SST.  The classes expect to
+// encapsulate a pointer to an object and a pointer to a member
+// function of the objects class.  The classes also allow you to
+// optionally add one additional piece of static data to be passed
+// into the callback function along with any data provided by the
+// caller.  There are two versions of this class, one that has no data
+// passed from the caller (ending with SSTHandler0), and one that has
+// a single item passed from the caller (SSTHandler1).
+
+// These classes provide the full functionality of the handlers and
+// can be added to a class with the "using" keyword, as follows (a
+// class can use any type name they'd like in place of HandlerBase and
+// Handler, though those names are preferred for consistency):
+
+// using HandlerBase = SSTHandlerBase<return_type_of_callback, arg_type_of_callback>;
+
+// template <typename classT, typename dataT = void>
+// using Handler = SSTHandler<return_type_of_callback, arg_type_of_callback classT, dataT>;
+
+// Or:
+
+// using HandlerBase = SSTHandlerBaseNoArgs<return_type_of_callback>;
+
+// template <return_type_of_callback, typename classT, typename dataT = void>
+// using Handler = SSTHandlerNoArgs<classT, dataT>;
```

Comment:
Should this be:
`using HandlerBase = SSTHandlerNoArgs<return_type_of_callback, classT, dataT>;`

Reply:
Not quite, but you are right that line 42 is wrong, it should just be "using Handler = ..." (not HandlerBase).  I'll fix it.